packages/noya-state/src/reducers/application.ts

```diff
@@ -249,59 +258,147 @@ export function reducer(
         state.selectedPage = action[1];
       });
     }
-    case 'alignLeft': {
+    case 'distribute': {
       const pageIndex = getCurrentPageIndex(state);
       const layerIndexPaths = getSelectedLayerIndexPaths(state);
+      const axis = action[1];
 
       return produce(state, (state) => {
+        const page = state.sketch.pages[pageIndex];
+        const selectedRect = getSelectedRect(layerIndexPaths, page);
         const layers = accessPageLayers(state, pageIndex, layerIndexPaths);
-        const [leftmostLayer] = layers.sort(
-          (layerA, layerB) => layerA.frame.x - layerB.frame.x,
+        const combinedWidths = layers.reduce(
```

Comment:
There may be a scaling issue here (which I describe below on `sortedLayerIndexPaths`) but I don't think we need to handle it yet